pkg/skr/gcpnfsvolume/deletePersistentVolumeClaim.go

```diff
@@ -0,0 +1,51 @@
+package gcpnfsvolume
+
+import (
+	"context"
+	"fmt"
+
+	cloudresourcesv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-resources/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func deletePersistentVolumeClaim(ctx context.Context, st composed.State) (error, context.Context) {
+	state := st.(*State)
+	logger := composed.LoggerFromCtx(ctx)
+
+	if !composed.MarkedForDeletionPredicate(ctx, st) {
+		return nil, nil
+	}
+
+	if state.PVC == nil {
+		return nil, nil
+	}
+
+	if !state.PVC.DeletionTimestamp.IsZero() {
+		return nil, nil
+	}
+
+	err, _ := composed.UpdateStatus(state.ObjAsGcpNfsVolume()).
+		SetCondition(metav1.Condition{
+			Type:    cloudresourcesv1beta1.ConditionTypeDeleting,
+			Status:  metav1.ConditionTrue,
+			Reason:  cloudresourcesv1beta1.ConditionReasonDeletingPVC,
+			Message: fmt.Sprintf("Deleting PersistentVolumeClaim %s", state.PVC.Name),
+		}).
+		ErrorLogMessage("Error setting ConditionReasonDeletingPVC condition on GcpNfsVolume").
+		SuccessErrorNil().
+		FailedError(composed.StopWithRequeue).
+		Run(ctx, state)
+	if err != nil {
+		return err, nil
+	}
+
+	logger.Info("Deleting PVC for GcpNfsVolume")
+
+	err = state.Cluster().K8sClient().Delete(ctx, state.PVC)
```

Comment:
How do we know that this PVC is not being used by any pods? Do we need to validate it?
What happens if pvc is referenced in pods, and the user called delete on the GcpNfsVolume?

Reply:
Finalizer from k8s will stop it from being deleted:
https://github.com/kubernetes/kubernetes/blob/d145bf09074fcd9b884daa582e0f19f60450a39d/pkg/controller/volume/pvcprotection/pvc_protection_controller.go